### ISSUE TITLE
Sparkline resizing on deployments card in real-time when window resized

### ISSUE THREAD

**Issue by c-koehler** [openshiftio/openshift.io#1859]
Resizing of the webpage when a deployment-card is open (with the sparkline graphs) has a delay for resizing until the polling causes it to get new graph data. This means when someone resizes a window, the text for the data (like _Memory: X megs_) will overlap with the chart, which doesn't look good.

A previous changeset mitigated this issue by destroying the chart on collapse and creating it when expanded, which was done to help save memory so the application didn't crash.

---

_The following might be a way to solve this issue:_

What I believe happens is new data comes in and the chart renders the new graph from the new data.

A way I came across that may be able to fix this (I'm unsure if it will) would be usage of the following function:
https://github.com/patternfly/patternfly-ng/blob/master/src/app/chart/chart.base.ts#L29

I haven't tested it myself since I was having trouble getting around the `protected` access modifier, @andrewazores noted that it should be able to be done. If it can, maybe it can be triggered for rendering again if the window size changes?

**Comment by c-koehler** [comment]
After discussion, the proposed idea of dynamically resizing may not be the best option, however I was curious if it worked and what kind of memory footprint it would leave with us if we have no choice but to go with it.

I made a test example here: https://github.com/c-koehler/fabric8-ui/commit/8b383bed5ded5798871936be0de87fb7ad280a52 (located on the following branch: https://github.com/c-koehler/fabric8-ui/tree/dynamic-resize)

What this will do is listen for resize events and refresh the chart as needed. It's a quick demo and designed only to show what can be done (it only targets one chart so you can see the comparison to the second one).

From doing it, it's generally smooth and gets all the resize cases. Trying to drag the window works pretty smoothly. The only weird thing is that when the developer tools comes in and out by pressing F12, sometimes it might not updated immediately, but for at least one of them it will properly adjust. The memory usage is interesting, dragging it back and forth uses up memory a lot but the GC seems to be collecting the memory and properly removing it. I noticed the memory usage would raise frequently when dragging (like 8.4%, 8.5%, 8.6%...), then after stopping the resizing it would return back to 8.4%. The constant linear growth from the data going into the chart is still there, and this would lead me to say that resizing and dragging around the window appears to be okay as the consumed memory is either garbage collected or freed at some point.

Also, while there is a TypeScript error because the code is calling a `protected` method, it will still compile and still works on my end, so it can be run and visualized.

After a discussion with @jiekang this method is considered to be a last resort. Hopefully it's a proof of concept that it can be done if there are no other options.

In summary:
- Doesn't appear to affect long term memory or aggravate the accumulating memory
- Resizes are very efficient and fast, looks smooth and nice
- Would still be in support of a better way of doing this than what was provided for this proof-of-concept

**Comment by aptmac** [comment]
Fixed in commit https://github.com/fabric8-ui/fabric8-ui/pull/2423
